src/javascripts/ng-admin/es6/lib/Entity/Entity.js

```diff
@@ -0,0 +1,178 @@
+import stringUtils from "../Utils/stringUtils";
+import Field from "../Field/Field";
+import DashboardView from '../View/DashboardView';
+import MenuView from '../View/MenuView';
+import ListView from '../View/ListView';
+import CreateView from '../View/CreateView';
+import EditView from '../View/EditView';
+import DeleteView from '../View/DeleteView';
+import ShowView from '../View/ShowView';
+
+class Entity {
+    constructor(name) {
+        this._name = name;
+        this._baseApiUrl = null;
+        this._label = null;
+        this._identifierField = new Field("id");
+        this._isReadOnly = false;
+        this._errorMessage = null;
+        this._order = 0;
+        this._url = null;
+
+        this._initViews();
+    }
+
+    get views() {
+        return this._views;
+    }
+
+    /** @deprecated Use corresponding view method instead (ie `dashboardView()` for instance) */
+    getViewByType(viewType) {
+        return this._views[viewType];
+    }
+
+    label() {
+        if (arguments.length) {
+            this._label = arguments[0];
+            return this;
+        }
+
+        if (this._label === null) {
+            return stringUtils.camelCase(this._name);
```

Comment:
You meant [Humanize](https://github.com/HubSpot/humanize) right? We got `humane` in our dependencies, but it is used for notifications (see [official site](http://wavded.github.io/humane-js/)).